### ISSUE TITLE
[format.parse.ctx] Paragraph 14 is gibberish

### ISSUE THREAD

**Issue by Eisenwave** [cplusplus/draft#6801]
> *Remarks*: Call expressions where `id >= num_args_` or the type of the corresponding format argument (after conversion to `basic_format_arg<Context>`) is not one of the types in `Ts...` are not core constant expressions ([expr.const]).

The meaning of this paragraph is completely lost on me:

1. What does it mean to say "is not one of the types in `Ts...` are not ..." ? This isn't valid English.
2. Why does this function have no *Effect* and no *Returns* paragraph? There are only *Remarks*, so I have no idea what it's actually meant to do.
3. What does "call expressions" even mean? Does it mean that this function calls expressions? Is it meant to say "Calls `expressions` and I've just missed some `expressions` definition? Does it mean that calling this function is a call expression?

Granted, I know very little about the formatting library but this wording just looks like gibberish to me.

**Comment by jwakely** [comment]
Call expressions where one of the following conditions is true are not core constant expressions ([expr.const]):

- id >= num_args_ 
- or the type of the corresponding format argument (after conversion to basic_format_arg<Context>) is not one of the types in Ts...

**Comment by jwakely** [comment]
It has no _Effects_, because it has no effects. The only effect is to be a core constant expression, or not. That's so that you get an error if the conditions are not met when the consteval function checks the format string.

**Comment by jwakely** [comment]
3. No, it doesn't mean that. It means "expressions that are called to this function"

**Comment by Eisenwave** [comment]
> 1. Read it as:
> 
> 
> Call expressions where one of the following conditions is true are not core constant expressions ([expr.const]):
> 
>   * id >= num_args_
>   * or the type of the corresponding format argument (after conversion to basic_format_arg) is not one of the types in Ts...

Yeah thanks, it took me a while. It's pretty late here and the run-on sentence just really messed with my head.



> It has no _Effects_, because it has no effects. The only effect is to be a core constant expression, or not. That's so that you get an error if the conditions are not met when the consteval function checks the format string. It has no _Returns_ because its return type is void.

I've read P2757R3 and it sort of makes sense, but it's quite the surprising function and I feel like this deserves a note which explains the intent at least.


> No, it doesn't mean that. It means "expressions that are calls of this function", i.e. invocations of this function.

I would have much preferred the wording "Invocations of this functions" over "Call expressions". It wasn't obvious to me that the very general statement about expressions is meant to apply to calls to this particular function. Maybe that's just me, but it stumped me. I can't think of having seen "Call expressions" used in such a way anywhere else in the standard either.

What do you think about the following wording?

> An invocation of this function is a core constant expression only if
> - `id < num_args_` is `true` and
> - the type of the corresponding format argument (after conversion to `basic_format_args<Context>`) is one the types in `Ts...`

**Comment by Eisenwave** [comment]
What do you think about the following wording?

> An invocation of this function is a core constant expression only if
> - `id < num_args_` is `true` and
> - the type of the corresponding format argument (after conversion to `basic_format_args<Context>`) is one the types in `Ts...`
